zerver/lib/actions.py

```diff
@@ -4979,7 +4987,7 @@ def gather_subscriptions_helper(user_profile: UserProfile,
     if user_profile.can_access_public_streams():
         never_subscribed_stream_ids = all_streams_id_set - sub_unsub_stream_ids
     else:
-        never_subscribed_stream_ids = set()
+        never_subscribed_stream_ids = set(web_public_stream_ids)
```

Comment:
Won't this potentially result in duplicates?  I think we probably want to change this to be `set(web_public_stream_ids - sub_unsubs_stream_ids`, right?